### PR TITLE
feat: add required check for object key not in properties

### LINKED ISSUE #225: Required property not set in properties
## 🐛 Bug Report

A required property not set in properties is not checked also if set as required. This whether it is set with additionalProperties to true and to false.

## To Reproduce

```js
test('object with required field not in properties schema', (t) => {
  const schema = {
    title: 'object with required field',
    type: 'object',
    properties: {
      num: {
        type: 'integer'
      }
    },
    additionalProperties: true,
    required: ['str']
  }
  const stringify = build(schema)

  try {
    stringify({})
    t.fail()
  } catch (e) {
    t.is(e.message, 'str is required!')
    t.pass()
  }
}
```

## Expected behavior

I suppose this test should pass (ref of the required property here https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.7).

Otherwise, since it is an edge case usually due to a badly generated schema, this behavior could also be documented.

## Your Environment

- *node version*: 12
- *os*: Mac

### PR DESCRIPTION
**Pull request by davidebianchi**
This PR closes #225

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -731,9 +735,33 @@ function buildCode (schema, code, laterCode, name, externalSchema, fullSchema) {
...
     }
   })
+
+  if (required && required.length > 0) {
+    code += 'var required = ['
+    for (var i = 0; i < required.length; i++) {
+      if (i > 0) {
+        code += ','
+      }
+      code += `'${required[i]}'`
```
Comment (mcollina) [inline index.js:745]: This should be properly escaped.

File: test/allof.test.js [hunk 2]
```diff
@@ -178,10 +178,10 @@ test('object with nested allOfs', (t) => {
       })
       t.is(value, '{"id1":1,"id2":2,"id3":3}')
     } catch (e) {
-      t.fail()
+      t.fail(e.message)
     }
   } catch (e) {
-    t.fail()
+    t.fail(e.message)
```
Comment (mcollina) [inline test/allof.test.js:184]: are those changes necessary?

**Comment by davidebianchi** [comment]
Actual benchmark does not have required key in schema (nor required schema not present in properties), so there is no change to the generated code actually tested.

File: test/allof.test.js (hunk 2 above)
Comment (davidebianchi) [reply test/allof.test.js]: No, I removed

File: index.js (hunk 1 above)
Comment (davidebianchi) [reply index.js]: I escaped using `$asString` function, is this the correct way to escape?
